src/controllers/module/youtubeModuleController.ts

```diff
@@ -0,0 +1,102 @@
+import {Request, Response, Router} from 'express'
+import {LearningCatalogue} from '../../learning-catalogue'
+import {ModuleFactory} from '../../learning-catalogue/model/factory/moduleFactory'
+import * as youtube from '../../lib/youtube'
+import {ContentRequest} from '../../extended'
+import {ModuleValidator} from '../../learning-catalogue/validator/moduleValidator'
+
+export class YoutubeModuleController {
+	learningCatalogue: LearningCatalogue
+	moduleValidator: ModuleValidator
+	moduleFactory: ModuleFactory
+	router: Router
+
+	constructor(learningCatalogue: LearningCatalogue, moduleValidator: ModuleValidator, moduleFactory: ModuleFactory) {
+		this.learningCatalogue = learningCatalogue
+		this.moduleValidator = moduleValidator
+		this.moduleFactory = moduleFactory
+		this.router = Router()
+
+		this.setRouterPaths()
+	}
+
+	private setRouterPaths() {
+		this.router.param('courseId', async (req, res, next, courseId) => {
+			const course = await this.learningCatalogue.getCourse(courseId)
+
+			if (course) {
+				res.locals.course = course
+				next()
+			} else {
+				res.sendStatus(404)
+			}
+		})
+
+		this.router.get('/content-management/courses/:courseId/add-youtube-module', this.getModule())
+		this.router.post('/content-management/courses/:courseId/add-youtube-module', this.setModule())
```

Comment:
remove the `add-` in your path as this would not be semantically correct when editing a YT module